### DISCUSSION TITLE
Why is my timezone not being set in Cypress with moment-timezone library?

### DISCUSSION THREAD

**Discussion by zubairrshahidd** [cypress-io/cypress#26839] [Questions and Help]
Need a way to set Browser's Timezone in Cypress

I have used moment-timezone library

1. Added in support>index.js `const moment = require('moment-timezone');`
2. Added TZ env variable in my cypress.config.js
```
"env": {
      "TZ": "America/New_York"
    },
```

3. set the timezone to in by beforeEach hook
```
beforeEach(() => {
 moment.tz.setDefault('America/New_York');
})
```
4. it block contains the assertion 
```
const browserTimeZone = moment.tz.guess(); 
expect(browserTimeZone).to.equal('America/New_York');
```

But it is not setting up the timezone, the assertion is failing
<img width="411" alt="Screenshot 2023-05-23 at 10 39 07 PM" src="https://github.com/cypress-io/cypress/assets/33728786/b92c4d08-7939-4241-a220-e91e2fd25eb2">

**Comment by portyu9** [comment]
Hi @zubairrshahidd 

`moment.tz.setDefault()` and `moment.tz.guess()` are measuring two different things.

This:

```js
moment.tz.setDefault('America/New_York')
```

changes the default timezone used by **new Moment instances**. It does **not** change the browser/OS timezone.

Meanwhile:

```js
moment.tz.guess()
```

asks Moment to detect the timezone of the JavaScript environment, normally via `Intl.DateTimeFormat().resolvedOptions().timeZone`.

Also:

```js
env: {
  TZ: 'America/New_York'
}
```

only creates a Cypress environment variable (`Cypress.env('TZ')`). It does not change the browser's timezone.

So this assertion is not expected to become true just because you called `setDefault()`:

```js
expect(moment.tz.guess()).to.equal('America/New_York')
```

If what you want to test is Moment's configured default, use something like:

```js
moment.tz.setDefault('America/New_York')

expect(moment().tz()).to.equal('America/New_York')
```

or, even better for deterministic tests, explicitly create values in the timezone being tested:

```js
const value = moment.tz('2023-05-23 12:00', 'America/New_York')

expect(value.tz()).to.equal('America/New_York')
```

If you specifically need to emulate the **browser's actual timezone**, that is a different requirement. Cypress still does not have a cross-browser `timezone` configuration option; Chromium timezone emulation can be done through the Chrome DevTools Protocol, but that is Chromium-specific rather than something `Cypress.env()` or Moment controls.

I also saw the screenshot attachment in the discussion, but it could not be fetched successfully, so I have not relied on anything visible only in that screenshot for this diagnosis.

Please mark this answer as accepted if it helped, thank you.
